storybookjs/storybook
# CSF-tools: Parse stories using typescript keywords 'satisfies' and 'as'

Closes #22422



## What I did


- Added if statements to retrieve correct node for story (similar to code for Meta node on [line 271](https://github.com/joaonunomota/storybook/blob/3d3a17b7cdb7fd7fcc7c2e631ee18a38878f8abe/code/lib/csf-tools/src/CsfFile.ts#L271) of CsfFile.ts)
- Added test for parsing stories using typescript `as` keyword
- Updated test for parsing stories using typescript `satisfies` keyword, it was a mislabelled duplicate of the "" test

## How to test

1. Run a sandbox for a typescript template, e.g. `yarn task --task sandbox --start-from auto --template react-vite/default-ts`
2. Update `@storybook/csf-tools` using `yarn build`
3. Update Button.stories.ts with the following:
```
export const Primary = {
  name: 'Primary (satisfies)',
  args: {
    primary: true,
    label: 'Button',
  },
} satisfies Story;

export const Secondary = {
  name: 'Secondary (as)',
  args: {
    label: 'Button',
  },
} as Story;
```
4. Open Storybook in your browser and confirm that they render correctly

![image](https://github.com/storybookjs/storybook/assets/16689477/f048de62-57ba-485d-8708-1b8d67180425)



## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers


`["cleanup", "BREAKING CHANGE", "feature request", "bug", "build", "documentation", "maintenance", "dependencies", "other"]`



### 🦋 Canary release



This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_
